providers/gcp/resources/access_approval.go

```diff
@@ -33,6 +33,15 @@ func (g *mqlGcpProject) accessApprovalSettings() (*mqlGcpAccessApprovalSettings,
 	}
 	id := g.Id.Data
 
+	serviceEnabled, err := g.isServiceEnabled(service_accessapproval)
+	if err != nil {
+		return nil, err
+	}
+	if !serviceEnabled {
+		g.AccessApprovalSettings.State = plugin.StateIsSet | plugin.StateIsNull
```

Comment:
I know in cases where we return nil and not error at some point further down the code, the provider panics. Is this the way to make sure MQL isn't trying to dereference a nil pointer?

Reply:
Yes, this prevents the panic because it sets the state explicitly.